databricks/koalas/series.py

```diff
@@ -4194,6 +4194,62 @@ def pct_change(self, periods=1):
 
         return self._with_new_scol((scol - prev_row) / prev_row)
 
+    def dot(self, other):
+        """
+        Compute the dot product between the Series and the columns of other.
+
+        This method computes the dot product between the Series and another
+        one, or the Series and each columns of a DataFrame.
+
+        It can also be called using `self @ other` in Python >= 3.5.
+
+        Parameters
+        ----------
+        other : Series, DataFrame.
+            The other object to compute the dot product with its columns.
+
+        Returns
+        -------
+        scalar, Series
+            Return the dot product of the Series and other if other is a
+            Series, the Series of the dot product of Series and each rows of
+            other if other is a DataFrame.
+
+        Notes
+        -----
+        The Series and other has to share the same index if other is a Series
+        or a DataFrame.
+
+        Examples
+        --------
+        >>> s = ks.Series([0, 1, 2, 3])
+
+        >>> s.dot(s)
+        14
+
+        >>> s @ s
+        14
+        """
+        if repr(self.index) != repr(other.index):
+            raise ValueError("matrices are not aligned")
+
+        if isinstance(other, DataFrame):
+            raise ValueError(
+                "Series.dot() is currently not supported with DataFrame since "
+                "it will cause expansive calculation as many as the number "
+                "of columns of DataFrame")
+
+        elif isinstance(other, Series):
```

Comment:
What happens if `other` is not `DataFrame` or `Series`? e.g., `int`.

Reply:
it raises `IndexError` in pandas like below

```python
>>> pser
0    0
1    1
2    2
3    3
Name: 0, dtype: int64

>>> pser.dot(2)
Traceback (most recent call last):
...
IndexError: tuple index out of range
```

i'll fix this for working same in koalas.